torch/_inductor/cudagraph_trees.py

```diff
@@ -349,14 +350,22 @@ def get_manager(
 
 
 def cudagraphify_impl(model, inputs, static_input_idxs, *args, **kwargs):
-    fn = None
+    fn_cache = {}
+
+    # Detect int inputs: we need to index on these
+    int_key = [i for i, v in enumerate(inputs) if isinstance(v, int)]
+    get_ints = operator.itemgetter(*int_key) if int_key else lambda _: None
+
     del inputs
 
     def deferred_cudagraphify(inputs):
```

Comment:
I don't think we're triggering a warmup for subsequent runs, since we just check if the function id has already been warmed up: https://github.com/pytorch/pytorch/blob/9f3c06e3f9ae67c06134b963de86e1196b96c04b/torch/_inductor/cudagraph_trees.py#L1728.

You can add a test to check that.

If you run with batch size 1, then batch size 8, the second run will do allocations that need to be warmed up. Similarly, if something like `cudnn.benchmark` is set to True, then we'd need to warmup on every iteration. For now it's safer to warmup every run. 

I think we want to add the cache key to checking we have warmed something up: https://github.com/pytorch/pytorch/blob/9f3c06e3f9ae67c06134b963de86e1196b96c04b/torch/_inductor/cudagraph_trees.py#L1621

Reply:
You allocate a new func id every time you add_function so I think it's fine. I'll test.

Reply:
Yup, confirmed with test.